python/Ganga/GPIDev/Lib/GangaList/GangaList.py

```diff
@@ -52,8 +52,8 @@ def makeGangaListByRef(_list, preparable=False):
     result = GangaList()
     if len(_list) == 0:
         return result
-    temp_list = [stripProxy(element) for element in _list]
-    result._list = temp_list
+
+    result._list.extend([stripProxy(element) for element in _list])
```

Comment:
@drmarkwslater This will not by design impact the write performance changes which turn off dirty cache flushing and turns it back on after a full ```__set__``` operation is complete.
Not unless you add something to the equivalent of:
```
for a in list:
  new_obj[a] = old_obj[a]
```

anything to the extent of:

```
newlist = oldlist
```

is fully atomic at the registry level (or really should be by now).


I'm very weary of something such as the code below failing:

```
## pseudo-code as this 'I think' works
job1.something = GangaList([elements])
job2.something.extend(job1.something)
p2 = job2.something[element_i].getJobObject()
p1 = job1.something[element_i].getJobObject()
assert p1 is not p2
```

I think we should be protected by the assignment operators now being correctly fixed to handle the parent but I'd like to have a test of what parent (and/or job object) each element has.